resteasy-core/src/main/java/org/jboss/resteasy/plugins/server/servlet/HttpServletResponseWrapper.java

```diff
@@ -276,10 +321,6 @@ public void onError(Throwable t)
       {
          synchronized (this) {
             asyncListenerCalled = true;
-            if (lastAsyncOperation != null) {
-               lastAsyncOperation.future.completeExceptionally(t);
-               lastAsyncOperation = null;
-            }
```

Comment:
I see what you are doing here. The only problem I see is if there are two operations in the queue one was successful but the future is not completed(as sos was not ready) and the other failed write with an exception, this logic will fail all the operations in the queue. This scenario will not matter for a chunked async write like in our case. But I am not sure if this will be a problem for SSE.

Reply:
Yes, that is true. I was wondering what might happen there. I guess this is part of the issue when trying to make non-reactive code reactive :)

Reply:
I've added this back as it does make sense to keep it.